app/upgrades/v17/upgrades_test.go

```diff
@@ -125,10 +142,24 @@ func (suite *UpgradeTestSuite) TestUpgrade() {
 					lastPoolID = poolID
 				}
 
-				return expectedCoinsUsedInUpgradeHandler, lastPoolID
+				existingPool := suite.PrepareConcentratedPoolWithCoins("ibc/1480B8FD20AD5FCAE81EA87584D269547DD4D436843C1D20F15E00EB64743EF4", "uosmo")
+
+				// create few TWAP records for the pools
+				t1 := dummyTwapRecord(existingPool.GetId(), time.Now().Add(-time.Hour*24), "ibc/1480B8FD20AD5FCAE81EA87584D269547DD4D436843C1D20F15E00EB64743EF4", "uosmo", sdk.NewDec(10),
+					sdk.OneDec().MulInt64(10*10),
+					sdk.OneDec().MulInt64(3),
+					sdk.ZeroDec())
```

Comment:
It would be great to have more than one twap record be present in the test.

In particular, at least 2 historical twap records

We should also make sure that other pools' twap records are not accidentally updated. This can be done by also setting up twaps for one other pool and making sure that all of its records are not updated.

Reply:
added 2 pools checks with different historical records here: https://github.com/osmosis-labs/osmosis/pull/5939/commits/4ea33fa8248b92b65c17dc271335010d91716985#diff-8d0951a4357ce3ef3f57168835dcc72078bd30337d8f9882c3b0d1e64a815e77R146

Reply:
All twap records are still at the same time. Please make copies for different times so that there are multiple historical records at these different times